c++warnings.markdown

```diff
@@ -31,3 +31,25 @@ See also:
 
 - <http://en.cppreference.com/w/cpp/memory/new/operator_new>
 - <http://en.cppreference.com/w/cpp/memory/new/operator_delete>
+
+## error: delete called on 'BaseClass' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
+
+Whenever a C++ class has virtual methods and it's therefore meant to
+be used polymorphically, it's good practice to make sure to provide a
+virtual destructor:
```

Comment:
True, but given that I added `-Werror=delete-non-virtual-dtor` in AliPhysics, people will not be able to compile their code in any case.